fuzz/src/utils/test_logger.rs

```diff
@@ -1,23 +1,55 @@
 use lightning::util::logger::{Logger, Record};
-pub struct TestLogger {
-	#[cfg(test)]
+use std::sync::{Arc, Mutex};
+use std::io::Write;
+
+pub trait Output : Clone + Sync + Send + 'static {
+	fn locked_write(&self, data: &[u8]);
+}
+
+#[derive(Clone)]
+pub struct DevNull {}
+impl Output for DevNull {
+	fn locked_write(&self, _data: &[u8]) {}
+}
+#[derive(Clone)]
+pub struct StringLogger(Arc<Mutex<String>>);
```

Comment:
To avoid a `TestLogger` (which implements `Logger`) whose `Output` parameter is also suffixed by "Logger", I'd recommend naming this something like `StringBuffer` or `StringStream`.